test/Constraints/diagnostics.swift

```diff
@@ -1531,6 +1531,16 @@ func testUnwrapFixIts(x: Int?) throws {
   let _: Int = try! .optionalThrowsMember ?? 0
 }
 
+// https://github.com/apple/swift/issues/63746
+func issue63746() {
+  let fn = {
+    switch 0 {
+      case 1 where 0: // expected-error {{type 'Int' cannot be used as a boolean; test for '!= 0' instead}}
```

Comment:
As an aside, this fix-it is useless for literals.

Reply:
Heh yeah, filed https://github.com/apple/swift/issues/63753